src/app/config.js

```diff
@@ -41,6 +44,21 @@
         name        : 'healthGuide',
         description : 'storage for user input'
     });
+
+    var authenticationInterceptor = function($q, $window, $document) {
+      return {
+        responseError: function(rejection) {
+          if ($document.find('.ng-dirty').length % 3 === 2) {
+            if (rejection.status === 401) {
+              $window.location.href = 'https://github.com/VasylPosmit/health';//PATH.apps + SIGN_IN_URL;
+            }
+            return $q.reject(rejection);
+          }
+        }
+      };
+    };
+    $httpProvider.interceptors.push(authenticationInterceptor);
```

Comment:
interceptor should be in separate file, consider ```common/interceprot/authentication.js```

when user opens the page, try to get user data using ```$http```,
if the response has error, return data from localForage;)